snapcraft/tests/test_parser.py

```diff
@@ -534,3 +535,53 @@ def test_missing_fields(self, mock_get, mock_get_origin_data):
         }
         main(['--debug', '--index', TEST_OUTPUT_PATH])
         self.assertEqual(0, _get_part_list_count())
+
+    @mock.patch('snapcraft.internal.parser._get_origin_data')
+    @mock.patch('snapcraft.internal.sources.get')
+    def test_parsed_output_matches_wiki_order(
+            self, mock_get, mock_get_origin_data):
+        _create_example_output("""
+---
+maintainer: John Doe <john.doe@example.com>
+origin: lp:snapcraft-parser-example
+description: example main
+project-part: main
+---
+maintainer: Jim Doe <jim.doe@example.com>
+origin: lp:snapcraft-parser-example
+description: example main2
+project-part: main2
+---
+maintainer: Jim Doe <jim.doe@example.com>
+origin: lp:snapcraft-parser-example
+description: example main2
+project-part: app1
```

Comment:
From my experimentation the order was not predictable.  It's my understanding that python dicts make no promises with regard to order.